metadata/reader.go

```diff
@@ -617,10 +617,6 @@ func (r *reader) OpenFile(id uint32) (File, error) {
 			return fmt.Errorf("failed to get file bucket %d: %w", id, err)
 		}
 		size, _ = binary.Varint(b.Get(bucketKeySize))
-		m, _ := binary.Uvarint(b.Get(bucketKeyMode))
-		if !os.FileMode(uint32(m)).IsRegular() {
-			return fmt.Errorf("%q is not a regular file", id)
-		}
```

Comment:
Why did this chunk of code get removed?

Reply:
This prevented calling `metadataReader.OpenFile` on a non-regular file. That's probably not the right thing to do because a `metadata.File` is the information about a file's uncompressed offset/size and tar header offset/size within the uncompressed layer. That applied to all files, not just regular files.

In a practical sense, the big issue is that `reader.Verify` can't verify directories without this change.